.github/workflows/ci.yml

```diff
@@ -35,28 +38,40 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         rust: [stable, 1.45.0, beta]
+        args: ["", "--all-features"] 
     steps:
     - name: Checkout repository
       uses: actions/checkout@v1
     - name: Install Rust
       uses: hecrj/setup-rust-action@v1
       with:
         rust-version: ${{ matrix.rust }}
-    - run: cargo build --verbose
-    - run: cargo test --verbose 2>&1
-  allfeatures:
-    name: allfeatures
+    - run: cargo build --verbose ${{ matrix.args }}
+    - run: cargo test --verbose ${{ matrix.args }} 2>&1
+
+  coverage:
+    name: coverage
+    needs: build
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-latest]
-        rust: [stable, 1.45.0, beta]
+        rust: [stable]
+        args: ["", "--all-features"] 
     steps:
     - name: Checkout repository
       uses: actions/checkout@v1
     - name: Install Rust
       uses: hecrj/setup-rust-action@v1
       with:
         rust-version: ${{ matrix.rust }}
-    - run: cargo build --verbose --all-features
-    - run: cargo test --verbose --all-features 2>&1
+    - run: cargo install cargo-tarpaulin
+    # use a long timeout with tarpaulin as UI tests have to docker pull the
+    # cypress.io image which can cause a test timeout with the default tarpaulin
+    # timeout of 1 minute.
```

Comment:
We should solve this by pre-pulling the docker image rather than increasing a timeout.